x-pack/legacy/plugins/apm/public/components/app/APMIndicesPermission/__test__/APMIndicesPermission.test.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { shallow } from 'enzyme';
+import { APMIndicesPermission } from '../';
+
+import * as hooks from '../../../../hooks/useFetcher';
+import {
+  expectTextsInDocument,
+  MockApmPluginContextWrapper,
+  expectTextsNotInDocument
+} from '../../../../utils/testHelpers';
+
+describe('APMIndicesPermission', () => {
+  it('returns empty component when api status is loading', () => {
+    spyOn(hooks, 'useFetcher').and.returnValue({
+      status: hooks.FETCH_STATUS.LOADING
+    });
+    const component = shallow(<APMIndicesPermission />);
+    expect(component.isEmptyRender()).toBeTruthy();
+  });
+  it('returns empty component when api status is pending', () => {
+    spyOn(hooks, 'useFetcher').and.returnValue({
+      status: hooks.FETCH_STATUS.PENDING
+    });
+    const component = shallow(<APMIndicesPermission />);
+    expect(component.isEmptyRender()).toBeTruthy();
+  });
+  it('renders missing permission page', () => {
+    spyOn(hooks, 'useFetcher').and.returnValue({
+      status: hooks.FETCH_STATUS.SUCCESS,
+      data: {
+        'apm-*': { read: false }
+      }
+    });
+    const component = render(
+      <MockApmPluginContextWrapper>
+        <APMIndicesPermission />
+      </MockApmPluginContextWrapper>
+    );
+    expectTextsInDocument(component, [
+      'Missing permissions to access APM',
+      'Dismiss',
+      'apm-*'
+    ]);
+  });
+  it('shows escape hatch button when at least one indice has read privileges', () => {
+    spyOn(hooks, 'useFetcher').and.returnValue({
+      status: hooks.FETCH_STATUS.SUCCESS,
+      data: {
+        'apm-7.5.1-error-000001': { read: false },
+        'apm-7.5.1-metric-000001': { read: false },
+        'apm-7.5.1-transaction-000001': { read: false },
+        'apm-7.5.1-span-000001': { read: true }
```

Comment:
Not that it matters much because this is a test, but this would (almost) always be in the form of a wildcard like
```suggestion
        'apm-7.5.1-span-*': { read: true }
```

So to avoid tricking people into believing that this points to a specific index we should use the wildcard format?